iot/manager/manager_test.go

```diff
@@ -52,54 +68,149 @@ func setup(m *testing.M) {
 	}
 	client = pubsubClient
 
-	pubsubUUID, err := uuid.NewRandom()
+	topicID = createIDForTest("topic")
+
+	topic, err := client.CreateTopic(ctx, topicID)
 	if err != nil {
-		log.Fatalf("Could not generate uuid: %v", err)
+		log.Fatalf("Could not create topic: %v", err)
 	}
-	topicID = "golang-iot-topic-" + pubsubUUID.String()
+	fmt.Printf("Topic created: %v\n", topic)
+	topicName = topic.String()
 
-	t, err := client.CreateTopic(ctx, topicID)
+	// Generate UUID v1 for registry used for tests
+	registryID = createIDForTest("registry")
+
+	_, err = createRegistry(os.Stdout, projectID, region, registryID, topicName)
 	if err != nil {
-		log.Fatalf("Could not create topic: %v", err)
+		log.Fatalf("Could not create registry: %v\n", err)
 	}
-	fmt.Printf("Topic created: %v\n", t)
 }
 
 func shutdown() {
 	ctx := context.Background()
 
 	t := client.Topic(topicID)
 	if err := t.Delete(ctx); err != nil {
-		log.Fatalf("Could not delete topic: %v", err)
+		log.Fatalf("Could not delete topic: %v\n", err)
 	}
 	fmt.Printf("Deleted topic: %v\n", t)
+
+	if _, err := deleteRegistry(os.Stdout, projectID, region, registryID); err != nil {
+		log.Fatalf("Could not delete registry: %v\n", err)
+	}
 }
 
-func TestSendCommand(t *testing.T) {
-	// Generate UUID v1 for test registry and device
-	registryUUID, _ := uuid.NewRandom()
-	deviceUUID, _ := uuid.NewRandom()
+func TestCreateRegistry(t *testing.T) {
+	testRegistryID := createIDForTest("registry")
+
+	testutil.Retry(t, 10, 10*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		registry, err := createRegistry(buf, projectID, region, testRegistryID, topicName)
+		if err != nil {
+			r.Errorf("Could not create registry: %v\n", err)
+
+		} else {
```

Comment:
Return early and drop the else.

```suggestion
			return
		}
```